### PR TITLE
[macOS] Refactor PlatformView mutators

### PR DESCRIPTION
**Pull request by cbracken**
This refactors `[FlutterMutatorView applyLayer:]` to split out orthogonal behaviour into smaller functions for better readability/maintainability.

Extracts:
* General transform composition to CATransformFromMutations
* Opacity handling
* MasterClip computation
* RoundedRectClip computation

This change is a cleanup change for readability and makes no semantic changes to how mutator views are applied. Existing unit tests in [FlutterMutatorViewTest.mm](https://github.com/flutter/engine/blob/main/shell/platform/darwin/macos/framework/Source/FlutterMutatorViewTest.mm) are expected to pass without changes.

This is refactoring prior to applying a fix for rotation handling.

Issue: https://github.com/flutter/flutter/issues/124490

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by cbracken** [comment]
For reviewers, it may be easier to look at each of the commits separately. Each one extracts one particular bit of logic.

**Review by cyanglaz — APPROVED** [review]
LGTM mod nits

File: shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm [hunk 1]
```diff
@@ -211,6 +189,116 @@ CGPathRef PathFromRoundedRect(const FlutterRoundedRect& roundedRect) {
...
+                                               affineTransform);
+      master_clip = CGRectIntersection(rect, master_clip);
+    }
+  }
+  return master_clip;
+}
+
+/// A rounded rectangle and transform associated with it.
+typedef struct {
+  FlutterRoundedRect rect;
```
[the remark this answers is not in the document]
Comment (cbracken) [reply shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm]: Good call. Done.

File: shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm [hunk 2]
```diff
@@ -211,6 +189,116 @@ CGPathRef PathFromRoundedRect(const FlutterRoundedRect& roundedRect) {
...
+/// Returns the clip rect generated by the intersection of clips in the mutations vector.
+CGRect MasterClipFromMutations(CGRect bounds, const MutationVector& mutations) {
+  // Master clip in global logical coordinates. This is intersection of all clip rectangles
+  // present in mutators.
+  CGRect master_clip = bounds;
+
+  // Create the initial transform.
+  CATransform3D transform = CATransform3DIdentity;
+  for (auto mutation : mutations) {
+    if (mutation.type == kFlutterPlatformViewMutationTypeTransformation) {
```
[the remark this answers is not in the document]
Comment (cbracken) [reply shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm]: Agreed; plus it's in our style guide. :) Done.

File: shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm [hunk 3]
```diff
@@ -211,6 +189,116 @@ CGPathRef PathFromRoundedRect(const FlutterRoundedRect& roundedRect) {
...
+  for (const auto& r : rounded_rects) {
+    CGAffineTransform inverse = CGAffineTransformInvert(r.transform);
+    // Transform master clip to clip rect coordinates and check if this view intersects one of the
+    // corners, which means we need to use path clipping.
+    CGRect localMasterClip = CGRectApplyAffineTransform(master_clip, inverse);
+
+    // Only clip to rounded rectangle path if the view intersects some of the round corners. If
+    // not, clipping to masterClip is enough.
+    if (RoundRectCornerIntersects(r.rect, ToFlutterRect(localMasterClip))) {
+      CGPathRef path = PathFromRoundedRect(r.rect);
```
[the remark this answers is not in the document]
Comment (cbracken) [reply shell/platform/darwin/macos/framework/Source/FlutterMutatorView.mm]: Agreed - I'll look at this in a future cleanup once the structural refactoring is done.

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Pull request merged**